amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -132,6 +134,10 @@ public void notify(RecordsDeliveryAck recordsDeliveryAck) {
                 triggeringFlow = evictAckedEventAndScheduleNextEvent(recordsDeliveryAck);
             } catch (Throwable t) {
                 errorOccurred(triggeringFlow, t);
+            } finally {
+                if(isAwaitingFinalAckForCurrentSubscription) {
+                    lockObject.notifyAll();
```

Comment:
This is critical line, worth adding comments explaining the design.

Reply:
Sure